steamtinkerlaunch

```diff
@@ -25776,8 +25776,9 @@ function checkSteamDeckDependencies {
 		local DEPFILENAME="$2"
 		local DEPDIR="$3"
 		local REPOURL="$4"
+		local CHECKCMD="$($DEPCMD &> /dev/null --version && echo "OK" || echo "NOK")"
 
-		if [ -f "$(command -v "$DEPCMD")" ]; then
+		if [ -f "$(command -v "$DEPCMD")" && "$CHECKCMD" -eq "OK" ]; then
```

Comment:
The `if` check isn't valid, it needs to be separated into two `[ ]` conditions. Also, you should use `=` instead of `-eq` when comparing strings, this `-eq` does actually flag up with `integer expression expected`.

ShellCheck is flagging them up as follows:

> `SC2107 (error): Instead of [ a && b ], use [ a ] && [ b ]`
> `SC2170 (error): Invalid number for -eq. Use = to compare as string (or use $var to expand as a variable).`

Both of these should be fixable with the following (splits the checks into two `[ ]` tests, and uses `=` for string comparison:

```
if [ -f "$( command -v "$DEPCMD" )" ] && [ "$CHECKCMD" = "OK" ]; then
```